packages/components/src/DateTimePickers/DateRange/Inputs/Inputs.component.js

```diff
@@ -0,0 +1,82 @@
+import React, { useContext, useEffect, useRef } from 'react';
+import omit from 'lodash/omit';
+import DebounceInput from 'react-debounce-input';
+
+import { DateRangeContext } from '../Context';
+import { START_DATE, END_DATE } from '../constants';
+import InputSizer from '../../shared/InputSizer';
+
+import Icon from '../../../Icon';
+
+import theme from './Inputs.scss';
+
+const OMIT_PROPS_INPUT = ['startInputRef', 'endInputRef'];
+
+function Inputs(props) {
+	const { startDate, endDate, inputManagement } = useContext(DateRangeContext);
+	const { placeholder, onChange, onFocus, focusedInput } = inputManagement;
+
+	let startDateInputRef = useRef(null);
+	let endDateInputRef = useRef(null);
+
+	useEffect(() => {
+		if (startDateInputRef && focusedInput === START_DATE) {
+			startDateInputRef.focus();
+		}
+		if (endDateInputRef && focusedInput === END_DATE) {
+			endDateInputRef.focus();
+		}
+	});
```

Comment:
I don't understand what this is supposed to do, and how focusedInput would be changed.
And you haven't passed any dependencies to your useEffect, which means it is executed at each render.

Reply:
will add the condition array.
but I need to put some words about this `focusedInput`. `focusedInput` can be 'startDate' or 'endDate', and it's part of the context value of DateRangeContext. It's used for moving focus to end date after start date is selected.

Reply:
I have a responsibility problem here. It's more the manager that should manage the focus between from/to, when the from is selected from picker.
Let's merge as it is for now, but when you create the `InputDateRange` component, let's merge it with this one. So `InputDateRange` will compose the manager, the 2 inputs, and the popover. You can then register the `to` input refs in the manager, and focus on it, on `from` picker selection.

Reply:
ok, noted